### PR TITLE
[FIXED JENKINS-48498] Add become and become-user options

### PR DESCRIPTION
**Pull request by michaelcresswell**
[FIXED JENKINS-48498](https://issues.jenkins-ci.org/browse/JENKINS-48498) Add become and become-user options since sudo is deprecated. Mention sudo deprecation in UI but leave in place for now for backwards compatibility

### CODE REVIEW

**Comment by michaelcresswell** [comment]
Emilio (@escoem), 
In the [Google Group](https://groups.google.com/forum/#!topic/jenkinsci-dev/uBxRvurfZsM ) Jean-Christophe (@jcsirot) mentioned you should be able to merge PRs. Do you have a chance to look at this (and PR 18)?
Thanks!

File: src/main/java/org/jenkinsci/plugins/ansible/AnsibleAdHocCommandBuilder.java [hunk 1]
```diff
@@ -80,16 +84,18 @@
 
     @Deprecated
     public AnsibleAdHocCommandBuilder(String ansibleName, String hostPattern, Inventory inventory, String module,
-                                      String command, String credentialsId, boolean sudo, String sudoUser, int forks,
-                                      boolean unbufferedOutput, boolean colorizedOutput, boolean hostKeyChecking,
-                                      String additionalParameters)
+                                      String command, String credentialsId, boolean become,  String becomeUser,
```
Comment (escoem) [inline src/main/java/org/jenkinsci/plugins/ansible/AnsibleAdHocCommandBuilder.java]: Why are you adding new parameters to a deprecated constructor? are required?

File: src/main/java/org/jenkinsci/plugins/ansible/AnsiblePlaybookBuilder.java [hunk 2]
```diff
@@ -85,9 +89,9 @@
 
     @Deprecated
     public AnsiblePlaybookBuilder(String ansibleName, String playbook, Inventory inventory, String limit, String tags,
-                                  String skippedTags, String startAtTask, String credentialsId, boolean sudo,
-                                  String sudoUser, int forks, boolean unbufferedOutput, boolean colorizedOutput,
-                                  boolean hostKeyChecking, String additionalParameters)
+                                  String skippedTags, String startAtTask, String credentialsId, boolean become,
+                                  String becomeUser, boolean sudo, String sudoUser, int forks, boolean unbufferedOutput,
```
Comment (escoem) [inline src/main/java/org/jenkinsci/plugins/ansible/AnsiblePlaybookBuilder.java]: Are also required? Maybe a migration test would be great (but not required as we also have another constructor with the databoundconstructor annotation)

**Comment by escoem** [comment]
@michaelcresswell I will wait for your comments before merging the PR, anyway none of my requests will block the merge. Thanks for your contribution.

**Comment by michaelcresswell** [comment]
@escoem I thought about adding a warning where you suggested but Ansible itself warns about the deprecated option when appropriate. The plugin doesn't specify the earliest supported core Ansible version so I didn't want to falsely warn anybody stuck using an older one.

I've reverted the constructor changes.

Thank you for the feedback.

**Pull request merged**